src/main.py

```diff
@@ -79,66 +47,11 @@ def setup_argparser() -> argparse.ArgumentParser:
     else:
         endpoint = args.e if args.e else "http://localhost:8000/"
-        report = {"endpoint": endpoint}
-
+        report = {"endpoint": endpoint, "available": True, "dictionaries": {}}
         api_instance = ApiClient(endpoint, args.api_key)
-        try:
-            dictionary_list = list_dictionaries(api_instance)
-            report["available"] = True
-            dictionaries = args.d if args.d else dictionary_list
-            report["dictionaries"] = {}
-
-            sys.stderr.write("Evaluating %d dictionaries\n" % len(dictionaries))
-
-            for dictionary in dictionaries:
-                sys.stderr.write("Evaluating %s" % dictionary)
-
-                for entry_metric in entry_metrics:
-                    entry_metric.reset()
-
-                dict_report = {}
-                report["dictionaries"][dictionary] = dict_report
-
-                metadata = Metadata(api_instance.about(dictionary))
-
-                if metadata.errors:
-                    dict_report["metadataErrors"] = metadata.errors
-                else:
-                    for metadata_metric in metadata_metrics:
-                        dict_report.update(metadata_metric.apply(metadata))
-
-                offset = 0
-                while offset <= max_entries:
-                    entries = api_instance.list(dictionary, limit=LIMIT, offset=offset)
-                    if not entries:
-                        break
-
-                    for entry in entries:
-                        offset += 1
-                        if offset > max_entries:
-                            break
-                        entry = Entry(entry)
-                        if entry.errors:
-                            if "entryErrors" not in dict_report:
-                                dict_report["entryErrors"] = []
-                            dict_report["entryErrors"].extend(entry.errors)
-                        else:
-                            entry_report(api_instance, dictionary, entry,
-                                         dict_report)
-
-                    sys.stderr.write(".")
-                    sys.stderr.flush()
-
-                    if len(entries) < LIMIT:
-                        break
-
-                sys.stderr.write("\n")
-                for entry_metric in entry_metrics:
-                    if entry_metric.result(): #TODO
-                        print(entry_metric, entry_metric.result())
-                        dict_report.update(entry_metric.result())
-
-        except RequestException as e:
-            report["available"] = False
+        edie = Edie(api_instance, metadata_metrics_evaluators=metadata_evaluators, entry_metrics_evaluators=entry_evaluators)
 
-        print(json.dumps(report))
+        dictionaries: [Dictionary] = edie.load_dictionaries(args.d)
+        metadata_report = edie.evaluate_metadata()
+        entry_report = edie.evaluate_entries(10)
+        logging.info(json.dumps(edie.evaluation_report()))
```

Comment:
This needs to be a print statement. The assumption is the output is written to STDOUT